From 2ac2f416072d9968a9ea2dd9f48168223a1147b0 Mon Sep 17 00:00:00 2001
From: Baruch Siach <baruch@tkos.co.il>
Date: Tue, 18 Jul 2017 14:09:06 +0300
Subject: [PATCH] Fix build with uClibc-ng

Add a local definition for IPPROTO_MH to fix build with uClibc-ng. As of
version 1.0.25, the uClibc-ng netinet/in.h header does not provide this
definition. The kernel provided definition is masked by libc-compat.h. Add
this local fix until uClibc-ng syncs netinet/in.h with glibc.

This partially reverts upstream commit a9ae195a2169 (xfrm: get #define's from
linux includes)

Signed-off-by: Baruch Siach <baruch@tkos.co.il>
---
Upstream status: non upstreamable; local uClibc-ng compatibility fix
---
 ip/xfrm.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ip/xfrm.h b/ip/xfrm.h
index 54d80ce5e949..50b85550fb45 100644
--- a/ip/xfrm.h
+++ b/ip/xfrm.h
@@ -30,6 +30,10 @@
 #include <linux/xfrm.h>
 #include <linux/ipsec.h>
 
+#ifndef IPPROTO_MH
+# define IPPROTO_MH     135
+#endif
+
 #define XFRMS_RTA(x)  ((struct rtattr*)(((char*)(x)) + NLMSG_ALIGN(sizeof(struct xfrm_usersa_info))))
 #define XFRMS_PAYLOAD(n) NLMSG_PAYLOAD(n,sizeof(struct xfrm_usersa_info))
 
-- 
2.13.2

